### PR TITLE
TP 11952 TAD ensure admins cannot approve unpublishable firms

### PR DESCRIPTION
**Pull request by thenapking**
[19952](https://maps.tpondemand.com/entity/11952-tad-fix-updatealgoliaindexjob-errors)
Following comments from QA Testing, I have updated the code to ensure admins cannot approve unpublishable firms, as this would lead to errors in the frontend.

### CODE REVIEW

**Comment by jordanlg27** [comment]
https://maps.tpondemand.com/entity/11955

**Comment by jordanlg27** [comment]
https://maps.tpondemand.com/entity/11951

**Comment by jordanlg27** [comment]
https://maps.tpondemand.com/entity/11952

**Comment by thenapking** [comment]
NB: the conflicting files will be resolved on merge, by taking the updates from this branch over the master branch.

File: spec/features/admin/firm_approval_spec.rb [hunk 1]
```diff
@@ -33,9 +35,16 @@ def given_i_am_at_the_admin_firms_index_page
   end
 
   def given_there_are_fully_registered_principal_users_with_advisers_and_offices
```
Comment (davidtrussler) [inline spec/features/admin/firm_approval_spec.rb:36]: @thenapking I'm getting a failure in this test when I run it locally. Not sure if it's a data issue or something. 
image

**Comment by thenapking** [comment]
Try it now.  I was trying to fix Rubocop and accidentally left .do in there!

**Comment by davidtrussler** [comment]
> Try it now. I was trying to fix Rubocop and accidentally left .do in there!

OK, looks better! I shall approve it.

**Review by davidtrussler — APPROVED** [review]
LGTM! 👍

**Pull request merged**